students/mattmaeda/session01/break_me.py

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/python
+
+def raise_name_error():
+    print(i_do_no_exist)
+
+
+def raise_type_error():
+    x = 1 + a
+
+
+def raise_syntax_error():
+    eval('x===x')
+
+
+def raise_attribute_error():
+    from time import fakemodule
+
+
+if __name__ == "__main__":
+    try:
+        raise_name_error()
+    except Exception as e:
+        print("#### Name Error ####")
+        print(str(e) + "\n")
+
+    try:
+        raise_type_error()
+    except Exception as e:
+        print("#### Type Error ####")
+        print(str(e) + "\n")
+
```

Comment:
So, this is a clever way to avoid commenting out the functions, but doing it this way masks the error that you are getting. If you run your raise_type_error function outside of your try, except block you will notice that you are actually getting a name error, because a has not assigned to anything. So, to use your method to avoid commenting out things, but still make sure you are getting the correct error is to use `except NameError as e:` (or whichever error you are testing for). Generally speaking, you never want to catch *all* exceptions, but only the ones that you know how to deal with.